Scripts/Runtime/Jobs/Collections/DynamicBufferSharedWriteUtil.cs

```diff
@@ -0,0 +1,171 @@
+using Anvil.Unity.DOTS.Util;
+using System;
+using Unity.Entities;
+using UnityEngine;
+
+namespace Anvil.Unity.DOTS.Jobs
+{
+    /// <summary>
+    /// Helper class for managing access control to a <see cref="DynamicBuffer{T}"/> that you want to
+    /// write to in parallel from multiple different types of jobs.
+    /// </summary>
+    /// <remarks>
+    /// Unity by default treats a parallel job (JobTypeA run in parallel) as one job that can run over multiple
+    /// threads at the same time. If you want to have more than one job such as JobTypeA and JobTypeB, Unity
+    /// will not allow it unless you specify the <see cref="NativeDisableContainerSafetyRestriction"/> on
+    /// your <see cref="DynamicBuffer{T}"/>.
+    ///
+    /// Doing so means that you can now write to the <see cref="DynamicBuffer{T}"/> from multiple job types
+    /// but it assumes that you have taken the necessary precautions to ensure your jobs can't conflict with
+    /// each other. Typically this means only writing to an index that matches your
+    /// <see cref="NativeSetThreadIndex"/> or <see cref="ParallelCollectionUtil.CollectionIndexForThread"/>
+    ///
+    /// Unity's internal read and write handles for scheduling jobs assumes that all jobs are either a shared
+    /// read or exclusive write. Since we want both job types to start at the same time, this
+    /// <see cref="DynamicBufferSharedWriteHandle{T}"/> facilitates getting that start point while working
+    /// nicely in tandem with Unity's built in system that know nothing about your shared writing.
+    ///
+    /// This can be accomplished with a <see cref="CollectionAccessController{TContext}"/> but it requires
+    /// ALL systems in the project who touch the <see cref="DynamicBuffer{T}"/> in question to use the
+    /// <see cref="CollectionAccessController{TContext}"/> which isn't always feasible.
+    /// </remarks>
+    public static class DynamicBufferSharedWriteUtil
+    {
+        //*************************************************************************************************************
+        // INTERNAL INTERFACES
+        //*************************************************************************************************************
+
+        internal interface IDynamicBufferSharedWriteHandle : IDisposable
+        {
+            void RegisterSystemForSharedWrite(ComponentSystemBase system);
+        }
```

Comment:
Because its only purpose it to allow for multiple of them to be stored in the lookup dictionary. 

`Dictionary<ComponentType, IDynamicBufferSharedWriteHandle>` - ALLOWED
`Dictionary<ComponentType, DynamicBufferSharedWriteHandle<T>>` - NOT ALLOWED

You'll never use it outside of that so it's internal and part of the lookups.

Reply:
Not relevant anymore after the refactor to Systems